tasks/05_participants/participants.py

```diff
@@ -0,0 +1,226 @@
+                df.columns = ["raw_participant", "p_address", "p_phone"]
+
+                return df.to_dict(orient="records")
+
+    except Exception as e:
+        print(f"Pandas table parse error: {e}")
+
+    # If no participants table, return empty list for testing purposes
+    return []
+
+
+def parse_participant(html_raw=str) -> list[dict]:
+    try:
+        pd_raw_dicts = pd_raw_participants(html_raw=html_raw)
+        raw_html_parse = html_raw_participants(html_str=html_raw)
+        # print('len(raw_html_parse):', len(raw_html_parse))
+        html_participants = html_parse_participant(raw_html_parse)
+        # print('len(html_participants)', len(html_participants))
+
+    except Exception as e:
+        print(f"Failed to parse participant: {e}")
+        print(html_raw)
+        pass
+
+    out_dict_list = []
+    for i in range(len(html_participants)):
+        temp_dict = pd_raw_dicts[i] | html_participants[i]
+        out_dict_list.append(temp_dict)
+    return out_dict_list
+
+
+def process_participants(connection: sql.db_cnx(), case_row):
+    curs = connection.cursor()
+    raw = case_row["raw_text"]
+    case_id = case_row["case_id"]
+    case_number = case_row["case_number"]
+
+    if db_config.db_type == "sqlite":
+        query = """INSERT INTO participants
+                    (case_id, p_name, p_kind, p_role, p_org, p_address, p_phone, raw_participant)
+                    VALUES (?, ?, ?, ?, ?, ?, ?, ?)
+                """
+    elif db_config.db_type == "postgresql":
+        query = """INSERT INTO participants
+                    (case_id, p_name, p_kind, p_role, p_org, p_address, p_phone, raw_participant)
+                    VALUES (%s, %s, %s, %s, %s, %s, %s, %s);
+                """
+    try:
+        for r in parse_participant(raw):
+            curs.execute(
+                query,
+                (
+                    case_id,
+                    r["p_name"],
+                    r["p_kind"],
+                    r["p_role"],
+                    r["p_org"],
+                    r["p_address"],
+                    r["p_phone"],
+                    r["raw_participant"],
+                ),
+            )
+
+    except Exception as e:
+        if db_config.db_type == "sqlite":
+            error_query = """INSERT INTO error_log (case_id, participants_parse_error)
+                    VALUES (?, ?)
+                """
+        elif db_config.db_type == "postgresql":
+            error_query = """INSERT INTO error_log (case_id, participants_parse_error)
+                    VALUES (%s, %s);
+                """
+        print(f"Error parsing participants from case: {case_id}, {case_number}.")
+        curs.execute(error_query, (case_id, True))
+        raise e
+
+    finally:
+        curs.close()
+        connection.commit()
+
+
+def add_participant_row(case_id: int, r: list):
+    # insert relevant info to participants table in the db
+    try:
+        if db_config.db_type == "sqlite":
```

Comment:
this `if/elif` block that sets the query can be moved above and outside of the `try/except` block, since it's unlikely to cause an exception